### PR TITLE
fix(vertexai): use gs:// in safety-settings-multimodal

### PR DESCRIPTION
**Pull request by eliben**
Also make it generally more consistent with the multimodal sample

## Checklist
- [x] I have followed [Contributing Guidelines from CONTRIBUTING.MD](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md)
- [x] **Tests** pass:   `go test -v ./..` (see [Testing](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#testing))
- [x] **Code formatted**:   `gofmt` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [x] **Vetting** pass:   `go vet` (see [Formatting](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/CONTRIBUTING.md#formatting))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [ ] This sample adds a new sample directory, and I updated the [CODEOWNERS file](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/CODEOWNERS) with the codeowners for this sample
- [ ] This sample adds a new **Product API**, and I updated the [Blunderbuss issue/PR auto-assigner](https://github.com/GoogleCloudPlatform/golang-samples/blob/main/.github/blunderbuss.yml) with the codeowners for this sample
- [ ] Please **merge** this PR for me once it is approved

### CODE REVIEW

File: vertexai/safety-settings-multimodal/safety-settings-multimodal.go [hunk 1]
```diff
@@ -77,43 +70,17 @@ func generateContent(w io.Writer, prompt []genai.Part, projectID, location, mode
...
-	res, err := http.Get(image)
-	if err != nil || res.StatusCode != 200 {
-		return nil, err
-	}
-	defer res.Body.Close()
-	data, err := io.ReadAll(res.Body)
+	res, err := model.GenerateContent(ctx, img, genai.Text(prompt))
 	if err != nil {
-		return nil, fmt.Errorf("unable to read from http: %v", err)
+		return fmt.Errorf("unable to generate contents: %v", err)
```
Comment (muncus) [inline vertexai/safety-settings-multimodal/safety-settings-multimodal.go:81]:
```suggestion
		return fmt.Errorf("unable to generate contents: %w", err)
```

File: vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go [hunk 2]
```diff
@@ -19,38 +19,31 @@ import (
...
-	temp := 0.8
 
-	cat, _ := partFromImageURL("https://storage.googleapis.com/cloud-samples-data/generative-ai/image/320px-Felis_catus-cat_on_snow.jpg")
-
-	// create a multipart (multimodal) prompt
-	prompt := []genai.Part{
-		genai.Text("say something nice about this "),
-		cat,
-	}
+	image := "https://storage.googleapis.com/cloud-samples-data/generative-ai/image/320px-Felis_catus-cat_on_snow.jpg"
```
Comment (muncus) [inline vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go:35]:
```suggestion
	image := "gs://cloud-samples-data/generative-ai/image/320px-Felis_catus-cat_on_snow.jpg"
```

File: vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go [hunk 3]
```diff
@@ -19,38 +19,31 @@ import (
...
 	}
 
 	var buf bytes.Buffer
-	if err := generateContent(&buf, prompt, projectID, location, model, float32(temp)); err != nil {
+	if err := generateMultimodalContent(&buf, prompt, image, projectID, location, model); err != nil {
 		t.Fatal(err)
 	}
 
-	if got := buf.String(); !strings.Contains(got, "generate-content response") {
+	if got := buf.String(); !strings.Contains(got, "generated response: ") {
```
Comment (muncus) [inline vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go:46]: optional: we may want to switch this to the common 'want vs. got' strategy, so we can see the generated outputs when the test fails.

**Review by muncus — APPROVED** [review]

File: vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go (hunk 3 above)
Comment (eliben) [reply vertexai/safety-settings-multimodal/safety-settings-multimodal_test.go:46]: Let's do it for all the tests together as a separate step

**Pull request merged**